### ISSUE TITLE
fix fuse tests

### ISSUE THREAD

**Issue by evgmik** [sahib/brig#78]
We observe that `fuse` test are occasionally fail. This was mentioned in several
followups to issues (for example #77).

As was summarized in that discussion the bug is actually in `go v1.9` and higher, which treats some polls as blocking. See discussion of this in
https://github.com/bazil/fuse/issues/264#issuecomment-727269770

On our sides it is broken because the same process manages `fuse` mount and do the file operations with it. Which occasionally triggers the infinite wait loop which freezes the test, and leaves a `defunct` test process and makes high `load` in linux.

Good news nothing is wrong with `fuse` layer, bad news we need to fix its tests.

`basil/fuse` suggest to run two separate fuse processes during tests.
https://github.com/bazil/fuse/commit/3f5ad9058d81bedd9d0181e676373e6b8053bb2a

**Comment by sahib** [comment]
Ah, searching GitHub could have saved me some time :smiley: - good find @evgmik!
But at least it was quite educational. I read the code behind ``os.Open()`` & with the explanation the behavior suddenly makes sense. You want to have a go on that ticket?

We should probably also verify that we never do any I/O ourselves in the same process and leave a big fat warning comment so nobody else tries to do that.

**Comment by evgmik** [comment]
Well, it took me half a day of digging through our code. I pretty much went same route as you. Sprinkle debug output everywhere and convinced myself that it works at our end. I did not went as far as reading `os.Open` though.  Then I decided to look for trouble spot elsewhere and find those fuse issues.

Yes, I am working on fuse test fixes. But I might yelp for help at some point. My first attempt to understand how to use `fuse` test facility is slow, but so far I am moving forward.
